60_networks.Rmd

```diff
@@ -0,0 +1,664 @@
+* `nodeSize = "eigenvector"`: Nodes are scaled by eigenvector centrality.
+* `nodeSizeSpread = 3`: Node sizes are more similar if the value is decreased. This argument (in combination with cexNodes) is useful to enlarge small nodes while keeping the size of big nodes. Default is 4.
+* `nodeColor = "cluster"`: Node colors represent clusters.
+* `edgeTranspHigh = 20`: Edges with high edge weight are given some transparency so that node labels overlapping the edges are still readable.
+* `cex...`: Scaling of the node sizes, labels, title, etc.
+
+**Note:** Nodes are sometimes placed too close together so that the labels overlap. You may need to play around with the repulsion until you find a value where the labels are legible, but also the clusters are still well recognizable.
+
+```{r network_plot_single_cluster, fig.width=13, fig.height=10}
+plot(peerj_netprops,
+     layout = "spring",
+     repulsion = 0.98,
+     rmSingles = TRUE,
+     labelScale = FALSE,
+     nodeSize = "eigenvector",
+     nodeSizeSpread = 3,
+     nodeColor = "cluster", 
+     edgeTranspHigh = 20,
+     title1 = "Network on genus level", 
+     showTitle = TRUE,
+     cexTitle = 2.3,
+     cexNodes = 2,
+     cexLabels = 1,
+     cexHubLabels = 1.3,
+     mar = c(1, 3, 4, 8))
+
+legend(0.7, 1.1, cex = 1.7, title = "estimated correlation:",
+       legend = c("+","-"), lty = 1, lwd = 3, col = c("#009900","red"), 
+       bty = "n", horiz = TRUE)
+```
+
+Since hubs are highlighted by default, we see at a glance that Escherichia, Citrobacter, and Salmonella are hub nodes. Strinking are also the strong associations between Achromobacter and Bhargavaea, between Citrobacter and Escherichia, and between Geobacillus and Virgibacillus, which we have already seen in the edge list.
+
+#### Highlight data features
+
+We now color nodes according to their phylum. The sizes of the nodes are scaled by the mclr-transformed abundances of the genera, since SPRING uses this transformation to account for compositionality.
+
+```{r network_plot_single_phylum, fig.width=15, fig.height=10}
+# Get phyla names
+taxtab <- as(tax_table(peerj_phy), "matrix")
+phyla <- gsub("p__", "", taxtab[, "phylum"])
+names(phyla) <- taxtab[, "genus"]
+
+# Assign low abundant phyla to a "Other" class
+low_abundant <- names(table(phyla)[table(phyla) <= 3])
+phyla[phyla %in% low_abundant] <- "Other"
+
+# Take a look at the table
+table(phyla)
+phyla <- as.factor(phyla)
+
+# Define phylum colors
+phylcol <- c("cyan", "blue3", "red", "lawngreen", "yellow", "deeppink", "orange")
+
+plot(peerj_netprops,
+     layout = "spring",
+     repulsion = 0.98,
+     rmSingles = TRUE,
+     labelScale = FALSE,
+     nodeSize = "mclr",
+     nodeColor = "feature", 
+     featVecCol = phyla, 
+     colorVec =  phylcol,
+     edgeTranspHigh = 20,
+     title1 = "Network on genus level", 
+     showTitle = TRUE,
+     cexTitle = 2.3,
+     cexNodes = 2,
+     cexLabels = 1,
+     cexHubLabels = 1.3,
+     mar = c(1, 10, 4, 6))
```

Comment:
If some of these numbers are function defaults, it is not necessary to show the call. This will simplify the example without changing the outcome